### PR TITLE
Travis Config

### PR DESCRIPTION
**Pull request by brb3**
Addresses #92 

We should add a few _basic_ tests in this PR just to have something passing, then add tests for pending PRs as applicable.

### CODE REVIEW

**Comment by agarzola** [comment]
Here’s an interesting test I’ve been thinking of: if the `author` field in a post’s front matter doesn’t have a member’s username (tying it to their profile), check that its value is not the same as an existing member’s full name. If it _is_, the test fails with a message indicating that the author field should point to that member’s username. Examples:

If a post’s front matter has this:
```
author: Ryan Maynard
```
this test fails with a message like:
>Author has a member profile. Use their username instead.

But a post with this in the front matter:
```
author: Testy McTestface
```
passes the test because none of the members in the site has _Testy McTestface_ as their full name.

**Comment by brb3** [comment]
Tests work (and pass)

File: test/testBlogPosts.rb [hunk 1]
```diff
@@ -0,0 +1,53 @@
+require 'test/unit'
+require 'jekyll'
+
+class TestBlogPosts < Test::Unit::TestCase
+
+  def test_authorship
+
```
Comment (agarzola) [inline test/testBlogPosts.rb]: Let’s remove this empty line.

File: test/testBlogPosts.rb [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    Dir.foreach('_posts/') do |f|
+      if f.chars[0] == '.'
+        next
+      end
+
+      yaml = get_front_matter("_posts/#{f}")
+      assert_not_nil(yaml)
+      assert_not_nil(yaml['author'])
+      post_authors << yaml['author']
+
```
Comment (agarzola) [inline test/testBlogPosts.rb]: Remove empty line.

File: test/testBlogPosts.rb [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+      assert_not_nil(yaml['name'])
+      author_name = f.sub!(/\.md/, '')
+      members[author_name] = yaml['name']
+    end
+
+    # Make sure that the authors each have a member page.
+    post_authors.each do |author|
+      assert_true(members.key?(author) || !members.has_value?(author))
+    end
+
```
Comment (agarzola) [inline test/testBlogPosts.rb]: Remove extra line.

File: test/testBlogPosts.rb [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+  def get_front_matter(filename)
+    file_contents = File.read(filename)
+    if file_contents =~ Jekyll::Document::YAML_FRONT_MATTER_REGEXP
+      data = SafeYAML.load(Regexp.last_match(1))
+      return data
+    end
+
+    return nil
+  end
+
```
Comment (agarzola) [inline test/testBlogPosts.rb]: Remove extra line.

**Comment by agarzola** [comment]
And add a section to the readme that informs people submitting pull requests that:
- there is a testing suite
- they should run it locally (ideally before committing their work, but at least before submitting a pr)
- the command for running tests is: `bundle exec rake test`

File: test/testBlogPosts.rb [hunk 5]
```diff
@@ -13,8 +12,8 @@ def test_authorship
       end
 
       yaml = get_front_matter("_posts/#{f}")
-      assert_not_nil(yaml)
-      assert_not_nil(yaml['author'])
+      assert_not_nil(yaml, "#{f} Does not contain YAML")
```
Comment (agarzola) [inline test/testBlogPosts.rb]: To be more precise:
>#{f} does not contain YAML font matter.

Note the lower case `d`.

File: test/testBlogPosts.rb [hunk 6]
```diff
@@ -13,8 +12,8 @@ def test_authorship
...
+      assert_not_nil(yaml['author'], "#{f} Does not contain an author")
```
Comment (agarzola) [inline test/testBlogPosts.rb]: lower case `d` here as well.

File: test/testBlogPosts.rb [hunk 7]
```diff
@@ -35,9 +34,8 @@ def test_authorship
 
     # Make sure that the authors each have a member page.
     post_authors.each do |author|
-      assert_true(members.key?(author) || !members.has_value?(author))
+      assert(members.key?(author) || !members.has_value?(author), "#{author} should have a member page, or is using the name of an existing member")
```
Comment (agarzola) [inline test/testBlogPosts.rb]: >#{author} already has a member profile. Please use the username associated with this member.

**Pull request merged**